pkg/sql/catalog/typedesc/virtual_table_record_type.go

```diff
@@ -0,0 +1,316 @@
+
+// Adding implements the Descriptor interface.
+func (v VirtualTableRecordType) Adding() bool {
+	v.panicNotSupported("Adding")
+	return false
+}
+
+// Dropped implements the Descriptor interface.
+func (v VirtualTableRecordType) Dropped() bool {
+	v.panicNotSupported("Dropped")
+	return false
+}
+
+// Offline implements the Descriptor interface.
+func (v VirtualTableRecordType) Offline() bool {
+	v.panicNotSupported("Offline")
+	return false
+}
+
+// GetOfflineReason implements the Descriptor interface.
+func (v VirtualTableRecordType) GetOfflineReason() string {
+	v.panicNotSupported("GetOfflineReason")
+	return ""
+}
+
+// DescriptorProto implements the Descriptor interface.
+func (v VirtualTableRecordType) DescriptorProto() *descpb.Descriptor {
+	v.panicNotSupported("DescriptorProto")
+	return nil
+}
+
+// GetReferencedDescIDs implements the Descriptor interface.
+func (v VirtualTableRecordType) GetReferencedDescIDs() (catalog.DescriptorIDSet, error) {
+	return v.desc.GetReferencedDescIDs()
+}
+
+// ValidateSelf implements the Descriptor interface.
+func (v VirtualTableRecordType) ValidateSelf(_ catalog.ValidationErrorAccumulator) {}
+
+// ValidateCrossReferences implements the Descriptor interface.
+func (v VirtualTableRecordType) ValidateCrossReferences(
+	_ catalog.ValidationErrorAccumulator, _ catalog.ValidationDescGetter,
+) {
+}
+
+// ValidateTxnCommit implements the Descriptor interface.
+func (v VirtualTableRecordType) ValidateTxnCommit(
+	_ catalog.ValidationErrorAccumulator, _ catalog.ValidationDescGetter,
+) {
+}
+
+// TypeDesc implements the TypeDescriptor interface.
+func (v VirtualTableRecordType) TypeDesc() *descpb.TypeDescriptor {
+	v.panicNotSupported("TypeDesc")
+	return nil
+}
+
+// HydrateTypeInfoWithName implements the TypeDescriptor interface.
+func (v VirtualTableRecordType) HydrateTypeInfoWithName(
+	ctx context.Context, typ *types.T, name *tree.TypeName, res catalog.TypeDescriptorResolver,
+) error {
+	if typ.IsHydrated() {
+		return nil
+	}
+	typ.TypeMeta.Name = &types.UserDefinedTypeName{
+		Catalog:        name.Catalog(),
+		ExplicitSchema: name.ExplicitSchema,
+		Schema:         name.Schema(),
+		Name:           name.Object(),
+	}
+	typ.TypeMeta.Version = uint32(v.desc.GetVersion())
+	for _, t := range typ.TupleContents() {
+		if t.UserDefined() {
+			if err := hydrateElementType(ctx, t, res); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// MakeTypesT implements the TypeDescriptor interface.
+func (v VirtualTableRecordType) MakeTypesT(
+	_ context.Context, _ *tree.TypeName, _ catalog.TypeDescriptorResolver,
+) (*types.T, error) {
+	return v.typ, nil
+}
+
+// HasPendingSchemaChanges implements the TypeDescriptor interface.
+func (v VirtualTableRecordType) HasPendingSchemaChanges() bool { return false }
+
+// GetIDClosure implements the TypeDescriptor interface.
+func (v VirtualTableRecordType) GetIDClosure() (map[descpb.ID]struct{}, error) {
+	ret := make(map[descpb.ID]struct{})
```

Comment:
Should `v.GetID()` be included in the ID closure? I'm guessing perhaps not, but I'm asking to make sure that this has been considered.